### PR TITLE
Add "Add data button" on all view pages

### LINKED ISSUE #7924: Add "Add data" button everywhere in data source view browsing, not only at "Connected data" level

### PR DESCRIPTION
**Pull request by tdraier**
fixes: [#7924](https://github.com/dust-tt/dust/issues/7924)

## Description

- Add button to add data to a data source view at every level of the datasource view. 
- Renamed to "Link data"
- Expand the selection tree if there's only one possible ds
- Display "data request" on the correct ds
- Layout issues in request data modal (added gap)

Screenshot 2024-10-07 at 16 06 17
Screenshot 2024-10-07 at 16 06 24
Screenshot 2024-10-07 at 16 06 33
Screenshot 2024-10-07 at 16 06 45
Screenshot 2024-10-07 at 16 06 51

## Deploy Plan

deploy front

### CODE REVIEW

**Comment by spolu** [comment]
Can we check the wording with @gabhubert (in case not already done). I think we haven't used Link anywhere yet so I'm a bit bearish on adding a new verb?

File: front/components/vaults/VaultManagedDatasourcesViewsModal.tsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ export default function VaultManagedDataSourcesViewsModal({
       }}
       hasChanged={hasChanged}
       variant="side-md"
-      title={`Add connected data to vault "${vault.name}"`}
+      title={`Link connected data to vault "${vault.name}"`}
```
Comment (flvndvd) [inline front/components/vaults/VaultManagedDatasourcesViewsModal.tsx:140]: As seen IRL, let's keep "Add".

File: front/components/vaults/EditVaultManagedDatasourcesViews.tsx [hunk 2]
```diff
@@ -221,7 +231,7 @@ export function EditVaultManagedDataSourcesViews({
         <p>You have no connection set up.</p>
       </Dialog>
       <Button
-        label="Add data from connections"
+        label="Link data from connections"
```
Comment (flvndvd) [inline front/components/vaults/EditVaultManagedDatasourcesViews.tsx:234]: 🔙

**Comment by gabhubert** [comment]
per IRL: I thiiiiink Link has other negative consequences here.

**Review by flvndvd — APPROVED** [review]

**Pull request merged**